### PR TITLE
EVA-3147: added properties for variant load and accession import jobs

### PR DESCRIPTION
**Pull request by nitin-ebi**

### CODE REVIEW

File: ebi_eva_common_pyutils/config_utils.py [hunk 1]
```diff
@@ -81,6 +81,18 @@ def get_accession_pg_creds_for_profile(profile_name: str, settings_xml_file: str
     return pg_url, pg_user, pg_pass
 
 
+def get_variant_pg_creds_for_profile(profile_name: str, settings_xml_file: str):
```
Comment (sundarvenkata-EBI) [inline ebi_eva_common_pyutils/config_utils.py:84]:
```suggestion
def get_job_tracker_creds_for_profile(profile_name: str, settings_xml_file: str):
```

File: ebi_eva_common_pyutils/ncbi_utils.py [hunk 2]
```diff
@@ -94,3 +94,23 @@ def retrieve_species_scientific_name_from_tax_id_ncbi(taxid):
...
+
+def get_species_name_from_ncbi(assembly_acc):
+    # We first need to search for the species associated with the assembly
+    assembly_dicts = get_ncbi_assembly_dicts_from_term(assembly_acc)
+    taxid_and_assembly_name = set([
+        (assembly_dict.get('taxid'), assembly_dict.get('assemblyname'))
+        for assembly_dict in assembly_dicts
+        if assembly_dict.get('assemblyaccession') == assembly_acc or
+           assembly_dict.get('synonym', {}).get('genbank') == assembly_acc
+    ])
```
Comment (sundarvenkata-EBI) [inline ebi_eva_common_pyutils/ncbi_utils.py:107]: Courtesy @apriltuesday for this tip!
```suggestion
taxid_and_assembly_name = {
        (assembly_dict.get('taxid'), assembly_dict.get('assemblyname'))
        for assembly_dict in assembly_dicts
        if assembly_dict.get('assemblyaccession') == assembly_acc or
           assembly_dict.get('synonym', {}).get('genbank') == assembly_acc
    }
```

File: ebi_eva_common_pyutils/config_utils.py (hunk 1 above)
Comment (nitin-ebi) [reply ebi_eva_common_pyutils/config_utils.py]: updated

File: ebi_eva_common_pyutils/spring_properties.py [hunk 3]
```diff
@@ -121,6 +123,95 @@ def get_accessioning_properties(self, *, instance=None, target_assembly=None,  f
...
+            'config.restartability.allow': 'false',
+            'config.db.read-preference': read_preference,
+
+            'logging.level.embl.ebi.variation.eva': 'DEBUG',
+            'logging.level.org.opencb.opencga': 'DEBUG',
+            'logging.level.org.springframework': 'INFO',
+
+            'spring.main.allow-bean-definition-overriding': 'true',
+            }
+
```
Comment (sundarvenkata-EBI) [inline ebi_eva_common_pyutils/spring_properties.py:164]: Some observations:

* Suggest re-using common properties as much as possible (for ex: mongo properties) and only put in the additional properties here.
* If you feel that there are some properties here that should belong in the common properties and might benefit other jobs as well, suggest adding it [here](https://github.com/EBIvariation/eva-common-pyutils/blob/2c7d5dbdb8567f4c602f6db8bdfe4dca2c771350/ebi_eva_common_pyutils/spring_properties.py#L55)

File: ebi_eva_common_pyutils/spring_properties.py [hunk 4]
```diff
@@ -121,6 +123,95 @@ def get_accessioning_properties(self, *, instance=None, target_assembly=None,  f
...
+            'output.dir': str(output_dir),
+            'output.dir.annotation': str(annotation_dir),
+            'output.dir.statistics': str(stats_dir),
+
+            'spring.jpa.database-platform': 'org.hibernate.dialect.PostgreSQL9Dialect',
+            'spring.jpa.properties.hibernate.jdbc.lob.non_contextual_creation': True,
+            'spring.jpa.properties.hibernate.temp.use_jdbc_metadata_defaults': False,
+            'statistics.skip': False
+        },
```
Comment (sundarvenkata-EBI) [inline ebi_eva_common_pyutils/spring_properties.py:213]: This [comment](https://github.com/EBIvariation/eva-common-pyutils/pull/36/files#r1180212580) applies here too...

**Review by sundarvenkata-EBI — CHANGES_REQUESTED** [review]
Looks good but might need more discussion...

**Review by tcezard — APPROVED** [review]

File: ebi_eva_common_pyutils/ncbi_utils.py [hunk 5]
```diff
@@ -94,3 +94,23 @@ def retrieve_species_scientific_name_from_tax_id_ncbi(taxid):
...
+    taxid_and_assembly_name = {
+        (assembly_dict.get('taxid'), assembly_dict.get('assemblyname'))
+        for assembly_dict in assembly_dicts
+        if assembly_dict.get('assemblyaccession') == assembly_acc or
+           assembly_dict.get('synonym', {}).get('genbank') == assembly_acc
+    }
+    # This is a search so could retrieve multiple results
+    if len(taxid_and_assembly_name) != 1:
+        raise ValueError(f'Multiple assembly found for {assembly_acc}. '
+                         f'Cannot resolve single assembly for assembly {assembly_acc} in NCBI.')
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply ebi_eva_common_pyutils/ncbi_utils.py]: corrected

File: ebi_eva_common_pyutils/spring_properties.py (hunk 3 above)
Comment (nitin-ebi) [reply ebi_eva_common_pyutils/spring_properties.py]: Reformatted the code to remove duplicates and reuse common properties

File: ebi_eva_common_pyutils/spring_properties.py (hunk 4 above)
Comment (nitin-ebi) [reply ebi_eva_common_pyutils/spring_properties.py]: reformatted the code to reuse common properties

File: ebi_eva_common_pyutils/spring_properties.py [hunk 6]
```diff
@@ -97,19 +129,17 @@ def _accessioning_properties(self, *, instance=''):
...
+    def get_accessioning_properties(self, *, instance=None, target_assembly=None, fasta=None, assembly_report=None,
                                     project_accession=None, aggregation='BASIC', taxonomy_accession=None,
                                     vcf_file='', output_vcf='', chunk_size=100):
         """Properties for accessioning pipeline."""
         return self._format(
-            self._common_properties(assembly_accession=target_assembly, read_preference='secondaryPreferred',
-                                    chunk_size=chunk_size),
-            self._accessioning_properties(instance=instance),
+            self._common_accessioning_clustering_properties(instance=instance, assembly_accession=target_assembly,
+                                                            read_preference='secondaryPreferred', chunk_size=chunk_size),
```
Comment (sundarvenkata-EBI) [inline ebi_eva_common_pyutils/spring_properties.py:142]: Clustering should be done with "primary" as the default read preference.

File: ebi_eva_common_pyutils/spring_properties.py [hunk 7]
```diff
@@ -52,42 +53,73 @@ def _format_str(string, param):
...
+            'spring.datasource.driver-class-name': 'org.postgresql.Driver',
+            'spring.datasource.tomcat.max-active': 3,
+            'spring.jpa.generate-ddl': 'true',
+
             'mongodb.read-preference': read_preference,
+
             'spring.main.web-application-type': 'none',
             'spring.main.allow-bean-definition-overriding': 'true',
-            'spring.jpa.properties.hibernate.jdbc.lob.non_contextual_creation': 'true',
+            'spring.jpa.properties.hibernate.jdbc.lob.non_contextual_creation': True,
```
Comment (sundarvenkata-EBI) [inline ebi_eva_common_pyutils/spring_properties.py:96]: Any reason why "True" has to be a boolean instead of a string?

**Review by sundarvenkata-EBI — APPROVED** [review]

**Comment by sundarvenkata-EBI** [comment]
Looks like the build failing is due to an unresolved issue with Github actions. See [here](https://github.com/actions/runner/issues/2468)

**Pull request merged**